### PR TITLE
implement filters and change driverdaoimpl

### PR DESCRIPTION
**Pull request by IwalkInSilence**

### CODE REVIEW

File: src/main/java/core/controllers/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("pwd");
+        try {
+            Driver driver = authService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("driver_id", driver.getId());
```
Comment (Artificial-Friend) [inline src/main/java/core/controllers/LoginController.java]: image

File: src/main/java/core/dao/jdbc/DriverDaoJdbcImpl.java [hunk 2]
```diff
@@ -102,8 +105,29 @@ private static Driver getDriver(ResultSet resultSet) throws SQLException {
...
         Driver driver = new Driver(name, licenseNumber);
+        driver.setLogin(login);
+        driver.setPassword(password);
         driver.setId(id);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT * FROM drivers WHERE login = ? AND deleted = false";
```
Comment (Artificial-Friend) [inline src/main/java/core/dao/jdbc/DriverDaoJdbcImpl.java:119]: Don't use star here, in future this table could have more fields that would be redundant in your case

File: src/main/java/core/web/filters/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/drivers/add")) {
+            chain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null) {
```
Comment (Artificial-Friend) [inline src/main/java/core/web/filters/AuthenticationFilter.java:39]: what if driver delete himself? Will he still be able to have access to site?

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+<h4 style="color:red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/login">
+    Login <input type="text" name="login" required>
+    Password <input type="text" name="pwd" required>
+    <button type="submit">Login</button>
+</form>
+</body>
+</html>
```
Comment (Artificial-Friend) [inline src/main/webapp/WEB-INF/views/login.jsp]: add empty line

File: src/main/java/core/web/filters/CarsForDriverFilter.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CarsForDriverFilter implements Filter {
```
Comment (Artificial-Friend) [inline src/main/java/core/web/filters/CarsForDriverFilter.java]: You know what to do

**Review by Artificial-Friend — CHANGES_REQUESTED** [review]

**Review by Artificial-Friend — APPROVED** [review]

File: src/main/java/core/security/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driverFromDB = driverService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect username or password"));
+        if (driverFromDB.getPassword().equals(password)) {
+            return driverFromDB;
+        }
+        throw new AuthenticationException("Incorrect username or password");
```
Comment (RomanDubovskyi) [inline src/main/java/core/security/AuthenticationServiceImpl.java]:
```suggestion
        Optional<Driver> driverFromDB = driverService.findByLogin(login);
        if (driverFromDB.isPresent() && driverFromDB.get().getPassword().equals(password)) {
            return driverFromDB;
        }
        throw new AuthenticationException("Incorrect username or password");
```

File: src/main/java/core/web/filters/AuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/drivers/add")) {
+            chain.doFilter(req, resp);
+            return;
```
Comment (RomanDubovskyi) [inline src/main/java/core/web/filters/AuthenticationFilter.java]: ideally, we can replace if-else construction with a Set, and  when you list or URLs grows larger it won't make this structure complicated

File: src/main/java/core/web/filters/AuthenticationFilter.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final Injector injector =
+            Injector.getInstance("core");
+    private static final String DRIVER_ID = "driver_id";
+    private final DriverService driverService =
```
Comment (RomanDubovskyi) [inline src/main/java/core/web/filters/AuthenticationFilter.java]: do we use it here?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/web/filters/AuthenticationFilter.java [hunk 9]
```diff
@@ -31,7 +27,10 @@ public void doFilter(ServletRequest servletRequest,
         HttpServletRequest req = (HttpServletRequest) servletRequest;
         HttpServletResponse resp = (HttpServletResponse) servletResponse;
         String url = req.getServletPath();
-        if (url.equals("/login") || url.equals("/drivers/add")) {
+        Set<String> permittedUrls = new HashSet<>();
+        permittedUrls.add("/login");
+        permittedUrls.add("/drivers/add");
```
Comment (RomanDubovskyi) [inline src/main/java/core/web/filters/AuthenticationFilter.java]: use method 'init' for creating and initializing set

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/web/filters/AuthenticationFilter.java (hunk 9 above)
Comment (IwalkInSilence) [reply src/main/java/core/web/filters/AuthenticationFilter.java]: sorry forget about your comment yesterday ...

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**